base/array.jl

```diff
@@ -87,7 +87,6 @@ UInt8
 eltype(::Type) = Any
 eltype(::Type{Any}) = Any
 eltype(::Type{Bottom}) = throw(ArgumentError("Union{} does not have elements"))
-eltype(t::DataType) = eltype(supertype(t))
```

Comment:
Was the linalg code somehow relying on this (rather weird) method?

Reply:
Yes; there are some more methods `eltype(::Type{A{B}})` that need to be changed to `Type{<:A{B}}`.